tls/_constructs.py

```diff
@@ -137,21 +144,24 @@
 
 ASN1Cert = Struct(
     "ASN1Cert",
-    UBInt32("length"),   # TODO: Reject packets with length not in 1..2^24-1
+    SizeWithin(UBInt32("length"),
+               min_size=1, max_size=2 ** 24 - 1),
     Bytes("asn1_cert", lambda ctx: ctx.length),
 )
 
 Certificate = Struct(
-    "Certificate",  # TODO: Reject packets with length > 2 ** 24 - 1
-    UBInt32("certificates_length"),
+    "Certificate",
+    SizeWithin(UBInt32("certificates_length"),
+               min_size=1, max_size=2 ** 24 - 1),
     Bytes("certificates_bytes", lambda ctx: ctx.certificates_length),
 )
 
 Handshake = Struct(
     "Handshake",
     UBInt8("msg_type"),
-    UBInt24("length"),  # TODO: Reject packets with length > 2 ** 24
```

Comment:
Ah, good point. In general, we were using OpenSSL and the Go TLS implementation (https://golang.org/pkg/crypto/tls/) as references for answering questions like these and resolving ambiguities (e.g. do what openssl does). May I suggest taking a look there to see if this is a valid concern for those implementations as well, and if there are any hints about what they do to detect and handle such cases?